Source/WebCore/page/Navigation.cpp

```diff
@@ -130,8 +134,30 @@ Navigation::Result Navigation::forward(Options&&, Ref<DeferredPromise>&& committ
     return result;
 }
 
-void Navigation::updateCurrentEntry(UpdateCurrentEntryOptions&&)
+// https://html.spec.whatwg.org/multipage/nav-history-apis.html#dom-navigation-updatecurrententry
+ExceptionOr<void> Navigation::updateCurrentEntry(JSDOMGlobalObject& globalObject, UpdateCurrentEntryOptions&& options)
 {
+    if (!window()->frame() || !window()->frame()->document())
+        return Exception { ExceptionCode::InvalidStateError };
+
+    auto current = currentEntry();
+    if (!current)
+        return Exception { ExceptionCode::InvalidStateError };
+
+    auto serializedState = SerializedScriptValue::create(globalObject, options.state, SerializationForStorage::Yes, SerializationErrorMode::Throwing);
```

Comment:
I think we may want to use Nonthrowing for now or do something testing if there was an exceptions. The way the code is now seems like we ignore the exception. Some FIXME is also an option.

Reply:
WPT has tests covering this and it works as expected.